Mathlib/Algebra/Group/Prod.lean

```diff
@@ -737,69 +716,39 @@ variable {P : Type*} [Monoid P]
   with a commutation assumption:
   `f.noncommCoprod g _ (p : M × N) = f p.1 * g p.2`.
   (Noncommutative case; in the commutative case, use `MonoidHom.coprod`.)-/
-@[to_additive
+@[to_additive (attr := simps)
     "Coproduct of two `AddMonoidHom`s with the same codomain,
     with a commutation assumption:
     `f.noncommCoprod g (p : M × N) = f p.1 + g p.2`.
     (Noncommutative case; in the commutative case, use `AddHom.coprod`.)"]
 def noncommCoprod : M × N →* P where
   toFun := fun mn ↦ (f mn.fst) * (g mn.snd)
-  map_mul' := fun mn mn' ↦ by
-    simp only [Prod.fst_mul, Prod.snd_mul, map_mul]
-    simp only [mul_assoc]
-    apply congr_arg₂ _ rfl
-    simp only [← mul_assoc]
-    apply congr_arg₂ _ _ rfl
-    apply comm
-  map_one' := by
-    simp only [Prod.fst_one, Prod.snd_one, map_one, mul_one]
-
-@[to_additive (attr := simp)]
-theorem noncommCoprod_apply (p : M × N) :
-  f.noncommCoprod g comm p = f p.1 * g p.2 :=
-  rfl
-
-/- -- simp_normal_form complains
-@[to_additive (attr := simp)]
-theorem noncommCoprod_apply' (p : M × N) :
-    f.noncommCoprod g comm p = g p.2 * f p.1 :=
-  comm p.1 p.2
--/
+  map_one' := by simp only [Prod.fst_one, Prod.snd_one, map_one, mul_one]
+  __ := f.toMulHom.noncommCoprod g.toMulHom comm
```

Comment:
This one is a pretty important trick; don't reprove the same thing again if you can re-use the existing structure from the `MulHom`.